pkg/controllers/workapplier/controller.go

```diff
@@ -496,18 +496,98 @@ func (r *Reconciler) garbageCollectAppliedWork(ctx context.Context, work *fleetv
 	appliedWork := fleetv1beta1.AppliedWork{
 		ObjectMeta: metav1.ObjectMeta{Name: work.Name},
 	}
-	err := r.spokeClient.Delete(ctx, &appliedWork, &client.DeleteOptions{PropagationPolicy: &deletePolicy})
-	switch {
-	case apierrors.IsNotFound(err):
-		klog.V(2).InfoS("The appliedWork is already deleted", "appliedWork", work.Name)
-	case err != nil:
-		klog.ErrorS(err, "Failed to delete the appliedWork", "appliedWork", work.Name)
+	if work.Spec.ApplyStrategy != nil {
+		if work.Spec.ApplyStrategy.AllowCoOwnership {
```

Comment:
i'm thinking of to keep the ux simple, no matter whether the resource is co-owned by others or not, we display such information to the user that the resources cannot be deleted.